### DISCUSSION TITLE
Download files on Android

### DISCUSSION THREAD

**Discussion by aprakash-work** [flyerhq/flutter_chat_ui#219] [Q&A] [answered]
I implemented file messages as described in the docs using file_picker and open_files. On iOs, I'm able to send a file message, open it, and save the file to the device. On Android, I don't see any option to save the file to the device. Am I missing something? Is this an open_files configuration? I can 'Share' and 'Save to Drive', but looking for how to save specifically to the device.

Appreciate any help, thanks.

**Comment by demchenkoalex** [comment, marked as answer]
Hey, yes, you will need to look into the files libraries. Docs are only giving a suggestion how you could work with files, other than that it is outside the scope of the chat itself. That was the idea when designing a chat, you could use any dependencies you want, chat is mostly dependency agnostic.
